kodilla-good-patterns/src/main/java/com/kodilla/good/patterns/food2door/data/Order.java

```diff
@@ -0,0 +1,126 @@
+package com.kodilla.good.patterns.food2door.data;
+
+import java.time.LocalDate;
+
+public final class Order {
+    private LocalDate date;
+    private final Company producer;
+    private final Company recipient;
+    private final String productName;
+    private final String measure;
+    private double quantity;
+    private final double price;
+    // derived
+    private final double amount;
+
+    public Order(final LocalDate  date, final Company producer, final Company recipient, final String productName,
+                 final String measure, final double quantity, final double price) {
+        validate(date, producer, recipient, productName, measure, quantity, price);
+
+        this.date = date;
+        this.producer = producer;
+        this.recipient = recipient;
+        this.productName = productName;
+        this.measure = measure;
+        this.quantity = quantity;
+        this.price = price;
+        this.amount = Math.round(price * quantity * 100) / 100;
+    }
+
+    private void validate(LocalDate  date, Company producer, Company recipient,
+                          String productName, String measure, double quantity,
+                          double price) {
+        if (date == null) {
+            throw new IllegalArgumentException("Date is null");
+        }
+        if (producer == null) {
+            throw new IllegalArgumentException("Producer is null");
+        }
+        if (recipient == null) {
+            throw new IllegalArgumentException("Recipient is null");
+        }
+        if (productName == null || productName.isEmpty()) {
```

Comment:
You can add static method in helper class which will simplify string validation - as you can notice similar code is repeated in many places. Another option is to use library which implements JSR-303, for example http://hibernate.org/validator/